src/libstd/net/ip.rs

```diff
@@ -0,0 +1,449 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use prelude::v1::*;
+
+use cmp::Ordering;
+use hash;
+use fmt;
+use libc;
+use sys_common::{AsInner, FromInner};
+use net::{hton, ntoh};
+
+/// Representation of an IPv4 address.
+#[derive(Copy)]
+pub struct Ipv4Addr {
+    inner: libc::in_addr,
+}
+
+/// Representation of an IPv6 address.
+#[derive(Copy)]
+pub struct Ipv6Addr {
+    inner: libc::in6_addr,
+}
+
+#[allow(missing_docs)]
+#[derive(Copy, PartialEq, Eq, Clone, Hash, Debug)]
+pub enum Ipv6MulticastScope {
```

Comment:
Yes, this set could grow. There's a possibility that IETF could add up to seven more address scopes. They've done so a couple of times since the list was created: [Realm-local was added last year](https://tools.ietf.org/html/rfc7346), and [RFC3513 added admin-local and renamed node-local to interface-local in 2003](https://tools.ietf.org/html/rfc3513).

There isn't any way to expand an enum without breaking non-"_ => ..." matches, is there? Could we add a warning that tells users to include a _ case to handle future expansion? If not, is there any other way to make this stable?